x-pack/filebeat/tests/integration/managerV2_test.go

```diff
@@ -500,6 +528,170 @@ func TestRecoverFromInvalidOutputConfiguration(t *testing.T) {
 	}
 }
 
+func TestAgentPackageVersionOnStartUpInfo(t *testing.T) {
```

Comment:
I may be missing it, but I don't see the test verifying that Beats set the new BuildHash field in CheckinObserved. I don't think anything is actually testing that yet.